addOns/reports/src/main/java/org/zaproxy/addon/reports/sarif/SarifRule.java

```diff
@@ -0,0 +1,159 @@
+
+    public String getName() {
+        return alert.getName();
+    }
+
+    public String getFullDescription() {
+        return alert.getDescription();
+    }
+
+    public String getShortDescription() {
+        return alert.getName();
+    }
+
+    public SarifRuleProperties getProperties() {
+        return ruleProperties;
+    }
+
+    @Override
+    public int compareTo(SarifRule o) {
+        return alert.getPluginId() - o.alert.getPluginId();
+    }
+
+    public List<SarifRuleRelationShip> getRelationShips() {
+        if (relationShips == null) {
+            relationShips = createRelationShips();
+        }
+        return relationShips;
+    }
+
+    private List<SarifRuleRelationShip> createRelationShips() {
+        List<SarifRuleRelationShip> list = new ArrayList<>();
+        /* CWE relationship */
+        if (alert.getCweId() > 0) {
+
+            SarifRuleRelationShip cweRelation = new SarifRuleRelationShip();
+            cweRelation.kinds.add("superset");
+            cweRelation.target.sarifGuid = SarifGuid.createCweGuid(alert.getCweId());
+            cweRelation.target.id = "" + alert.getCweId();
+            cweRelation.target.toolComponent = SarifToolData.INSTANCE.getCwe();
+
+            list.add(cweRelation);
+        }
+        return list;
+    }
+
+    public class SarifRuleRelationShip {
+        List<String> kinds = new ArrayList<>();
+        SarifRuleRelationShipTarget target = new SarifRuleRelationShipTarget();
+
+        public SarifRuleRelationShipTarget getTarget() {
+            return target;
+        }
+
+        public List<String> getKinds() {
+            return kinds;
+        }
+    }
+
+    public class SarifRuleRelationShipTarget {
+        SarifToolComponent toolComponent;
+        SarifGuid sarifGuid;
+        String id;
+
+        public SarifToolComponent getToolComponent() {
+            return toolComponent;
+        }
+
+        public String getGuid() {
+            return sarifGuid.getGuid();
+        }
+
+        public String getId() {
+            return id;
+        }
+    }
+
+    public class SarifRuleProperties {
+        private SarifMessage solution;
+        private List<String> references;
+
+        public Collection<String> getReferences() {
+            return references;
+        }
+
+        public SarifMessage getSolution() {
+            return solution;
+        }
+
+        public String getConfidence() {
+            switch (alert.getConfidence()) {
+                case Alert.CONFIDENCE_FALSE_POSITIVE:
+                    return "false-positive";
+                case Alert.CONFIDENCE_MEDIUM:
+                    return "medium";
+                case Alert.CONFIDENCE_HIGH:
+                    return "high";
+                case Alert.CONFIDENCE_LOW:
+                    return "low";
+                case Alert.CONFIDENCE_USER_CONFIRMED:
+                    return "confirmed";
+                default:
+                    return "unknown";
+            }
```

Comment:
i18n?

Reply:
SARIF has no official confidence element but there is a general `properties` element which can be used by vendors/products to add additional content. 

So I decided to add `confidence` to the report by using `properties` element.

About the I18n: I do not want to add i18n here because
- Those values shall be readable by a machine
- We would need to add something like a "confidence-language" property as well to describe which language is used  and maybe something like a  "confidence-i18n" propertty in addition to have it still machine readable by another product (in our case SecHub)
- Properties are just a key,value map and there is no easy possibility to describe a structure (to avoid the former mentioned additional property keys)
- Avoid effort in transalations
- Keep it simple